fdbserver/Status.actor.cpp

```diff
@@ -366,23 +372,30 @@ static StatusObject machineStatusFetcher(WorkerEvents mMetrics, vector<std::pair
 				statusObj["network"] = networkObj;
 
 				if (configuration.present()){
-					statusObj["excluded"] = true; // Will be set to false below if this or any later process is not excluded
+					notExcludedMap[machineId] = true; // Will be set to false below if this or any later process is not excluded
 				}
 
-				statusObj["contributing_workers"] = 0;
-
-				machineMap[machineId] = statusObj;
+				workerContribMap[machineId] = 0;
+				machineJsonMap[machineId] = statusObj;
 			}
-			if (configuration.present() && !configuration.get().isExcludedServer(it->first))
-				machineMap[machineId].get_obj()["excluded"] = false;
 
-			machineMap[machineId].get_obj()["contributing_workers"] = machineMap[machineId].get_obj()["contributing_workers"].get_int() + 1;
+			if (configuration.present() && !configuration.get().isExcludedServer(it->first))
+				notExcludedMap[machineId] = false;
+			workerContribMap[machineId] ++;
 		}
 		catch (Error& e) {
 			++failed;
 		}
 	}
 
+	// Add the status json for each machine with tracked values
+	for (auto mapPair : machineJsonMap) {
+		auto& machineId = mapPair.first;
+		machineJsonMap[machineId]["excluded"] = notExcludedMap[machineId];
+		machineJsonMap[machineId]["contributing_workers"] = workerContribMap[machineId];
+		machineMap[machineId] = machineJsonMap[machineId];
+	}
```

Comment:
consider only calling machineJsonMap[machineId] once and using for all three of these calls